### DISCUSSION TITLE
ClassVar without parameters

### DISCUSSION THREAD

**Discussion by JelleZijlstra** [python/typing#973] [General]
I found this code in Black:

```
class CustomSplitMapMixin:
    _Key: ClassVar = Tuple[StringID, str]
    _CUSTOM_SPLIT_MAP: ClassVar[Dict[_Key, tuple]] = {}
```

(simplified from https://github.com/psf/black/blob/main/src/black/trans.py#L207)

PEP 526 doesn't say anything about what `ClassVar` without arguments means, but mypy and pyright both accept this code. Should they?

**Comment by hauntsaninja** [comment]
I think so? There's an analogy to Final to be made and PEP 591 explicitly allows naked Finals. In particular, `x: ClassVar = y` seems like it should be strictly better for type checking than `x = y` in a way that "gradual typing" generally allows.

The black code looks like it just wants a class-scoped type alias, which we recently had a big discussion about on typing-sig.

**Comment by srittau** [comment]
I don't see the harm in allowing it, but if it's allowed, at least the documentation at https://docs.python.org/3/library/typing.html should be updated.
